### PR TITLE
Move to esbuild instead of browserify

### PR DESCRIPTION
**Pull request by relu91**
This PR should eventually completely move us from the old browserify build to a more modern toolchain based on esbuild and web/test-runner. There are still things to fix:
- [x] Rebase on main
- [x] Create a new workflow only for browser tests (we don't need the matrix for browser testing)
- [x] Configure web/test-runner to run on CI with the playwright or similar but use developer browser locally (?). 
- [x] fix code style errors in web test runner

### CODE REVIEW

**Comment by relu91** [comment]
I opted out browser tests by removing the `test` script in the browser bundle. So now the CI will only run node tests and in the next commit, I will create a workflow just for browser tests. The only incovient is that now developers should run `npm run test:all` instead of `npm run test`. `test:all` is automatically executed once you push too.

**Comment by danielpeintner** [comment]
> The only incovient is that now developers should run `npm run test:all` instead of `npm run test`. `test:all` is automatically executed once you push too.

We don't mention the test commands at all (at the moment). Since the CI will run the check, developers will notice problems latest once pushing. Hence, I don't think we _need_ to mention `test:all` (but we can of course).

**Comment by relu91** [comment]
Sorry there were complications with prettier (which I've updated to keep the new runner tests in check). I had to also re-run the format command since it seems that the new prettier has different rules. Sorry for making reviewing this harder... please let me know if there something wrong that you spot.

Regarding how the browser tests are run, they can run using `npm run test:browser` but only if the developer really want to test there. The new workflow should be activated once we merge this PR and should guarantee that even if the developer forgets we get always green light from browser too.

File: .github/workflows/browser-test.yml
Comment (danielpeintner) [inline .github/workflows/browser-test.yml:6]:
```suggestion
        branches: [master]
```

File: .github/workflows/browser-test.yml
Comment (danielpeintner) [inline .github/workflows/browser-test.yml:8]:
```suggestion
        branches: [master]
```

**Review by danielpeintner — CHANGES_REQUESTED** [review]
I didn't spot any obvious issue apart from "master" vs "main".

After fixing that, I am fine with merging...

Thanks a lot @relu91 !!

**Comment by relu91** [comment]
> I didn't spot any obvious issue apart from "master" vs "main".
> 
> After fixing that, I am fine with merging...
> 
> Thanks a lot @relu91 !!

You are right! that's why they didn't run!!! nice catch. pushed the fix.

**Comment by relu91** [comment]
Now we hit a bug with optional dependencies: 
> Cannot find module @rollup/rollup-linux-x64-gnu. npm has a bug related to optional dependencies (https://github.com/npm/cli/issues/4828).

This [error](https://github.com/eclipse-thingweb/node-wot/actions/runs/12317389224/job/34379768120?pr=1336#step:6:17) is suggesting to re-generate the `package-lock.json`

**Review by danielpeintner — APPROVED** [review]
I think now everything seems to work 👍

**Pull request merged**
